{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Connecting to patent db...\n",
      "Complete.\n"
     ]
    }
   ],
   "source": [
    "from sqlalchemy import create_engine\n",
    "import pandas as pd\n",
    "#sqlalchemy connection object\n",
    "print(\"Connecting to patent db...\")\n",
    "engine = create_engine('mysql+pymysql://ipmaster:oroonoko64@ipinformatics.ccmbmjzurfjl.us-west-1.rds.amazonaws.com:3306/ipinformatics')\n",
    "connection = engine.connect()\n",
    "print(\"Complete.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading in patent IDs...\n",
      "Complete.\n"
     ]
    }
   ],
   "source": [
    "#read in patent ids\n",
    "with open(\"patent_ids_705_small.csv\", \"r\") as f:\n",
    "    print(\"Reading in patent IDs...\")\n",
    "    idlist = f.read().splitlines()\n",
    "    print(\"Complete.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total chunks to do: 1\n",
      "Size of remainder: 22\n"
     ]
    }
   ],
   "source": [
    "#chunker\n",
    "chunks = []\n",
    "subchunk = []\n",
    "\n",
    "maxchunks = int(len(idlist)/5000)\n",
    "maxchunks_real = maxchunks + 1\n",
    "print(\"Total chunks to do: \" + str(maxchunks_real))\n",
    "\n",
    "remainder = len(idlist) - (maxchunks*5000)\n",
    "print(\"Size of remainder: \" + str(remainder))\n",
    "\n",
    "count = 0\n",
    "\n",
    "for id in idlist:\n",
    "    subchunk.append(id)\n",
    "    if count <= maxchunks:\n",
    "        if len(subchunk) == 5000:\n",
    "            yy = ','.join(str(s) for s in subchunk)\n",
    "            chunks.append(yy)\n",
    "            count = count + 1\n",
    "            subchunk = []\n",
    "    elif count > maxchunks:\n",
    "        if len(subchunk) == remainder:\n",
    "            yy = ','.join(str(s) for s in subchunk)\n",
    "            chunks.append(yy)\n",
    "            count = count + 1\n",
    "            subchunk = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  application_id filing_date_x patent_no_x grant_date_x  \\\n",
      "0       08681712    1996-07-29     5819234   1998-10-06   \n",
      "1       08879736    1997-06-20     6012040   2000-01-04   \n",
      "2       08960665    1997-10-30     5987427   1999-11-16   \n",
      "3       08892306    1997-07-14     5907617   1999-05-25   \n",
      "4       09775787    2001-02-03     6684119   2004-01-27   \n",
      "\n",
      "                                           company_x filing_date_y  \\\n",
      "0  The Chase Manhattan Bank                      ...    1996-07-29   \n",
      "1                                    NCR Corporation    1997-06-20   \n",
      "2                                    NCR Corporation    1997-10-30   \n",
      "3                                Digital River, Inc.    1997-07-14   \n",
      "4                                 Ford Motor Company    2001-02-03   \n",
      "\n",
      "  patent_no_y grant_date_y                                          company_y  \n",
      "0     5819234   1998-10-06  The Chase Manhattan Bank                      ...  \n",
      "1     6012040   2000-01-04                                    NCR Corporation  \n",
      "2     5987427   1999-11-16                                    NCR Corporation  \n",
      "3     5907617   1999-05-25                                Digital River, Inc.  \n",
      "4     6684119   2004-01-27                                 Ford Motor Company  \n"
     ]
    }
   ],
   "source": [
    "#Lookup Query No. 1\n",
    "\n",
    "lookup = \"SELECT s_no as application_id, filing_date, patent_no, date as grant_date, organization as company FROM iplogic.pair_application INNER JOIN ipinformatics.patentsview_patent ON iplogic.pair_application.patent_no = ipinformatics.patentsview_patent.id INNER JOIN ipinformatics.patentsview_raw_assignee ON iplogic.pair_application.patent_no = ipinformatics.patentsview_raw_assignee.patent_id WHERE iplogic.pair_application.patent_no IN (\"\n",
    "\n",
    "count = 0\n",
    "#df = pd.DataFrame(index =['application_id'], columns = ['application_id', 'filing_date', 'patent_no', 'grant_date', 'company'])\n",
    "\n",
    "for c in chunks:\n",
    "    count = count + 1\n",
    "    print(\"Chunk No. \" + str(count))\n",
    "    query = lookup + str(c) + \")\"\n",
    "    lookup_frame = pd.read_sql_query(query, connection)\n",
    "    df1 = lookup_frame\n",
    "    df = df1.merge(lookup_frame, on='application_id', how='left')\n",
    "    \n",
    "print(df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  application_id filing_date_x patent_no_x grant_date_x  \\\n",
      "0       08681712    1996-07-29     5819234   1998-10-06   \n",
      "1       08879736    1997-06-20     6012040   2000-01-04   \n",
      "2       08960665    1997-10-30     5987427   1999-11-16   \n",
      "3       08892306    1997-07-14     5907617   1999-05-25   \n",
      "4       09775787    2001-02-03     6684119   2004-01-27   \n",
      "\n",
      "                                           company_x filing_date_y  \\\n",
      "0  The Chase Manhattan Bank                      ...    1996-07-29   \n",
      "1                                    NCR Corporation    1997-06-20   \n",
      "2                                    NCR Corporation    1997-10-30   \n",
      "3                                Digital River, Inc.    1997-07-14   \n",
      "4                                 Ford Motor Company    2001-02-03   \n",
      "\n",
      "  patent_no_y grant_date_y                                          company_y  \n",
      "0     5819234   1998-10-06  The Chase Manhattan Bank                      ...  \n",
      "1     6012040   2000-01-04                                    NCR Corporation  \n",
      "2     5987427   1999-11-16                                    NCR Corporation  \n",
      "3     5907617   1999-05-25                                Digital River, Inc.  \n",
      "4     6684119   2004-01-27                                 Ford Motor Company  \n"
     ]
    }
   ],
   "source": [
    "df.drop(['filing_date_y','patent_no_y','grant_date_y','company_y'], axis=1)\n",
    "print(df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "lookup_frame.to_json(\"lookup.json\", orient='records', lines=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chunk No. 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D345393'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D341413'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D319145'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D297876'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'H0005487'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D297744'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'H0005460'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D362850'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D298137'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D344016'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D301202'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'PP4542'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D298410'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D310355'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D354155'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D350708'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D343629'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'H0010391'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D311833'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D307742'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D298066'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D325210'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D356054'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D333772'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D324673'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D323883'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'PP6341'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D305628'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D348634'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'PP6321'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D299182'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D264087'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D312594'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'H0015199'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D244359'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'H0004430'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE29149'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D245185'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'H0015679'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28965'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'H0003468'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28712'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE29914'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28685'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE29086'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D254870'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D281552'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D257180'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE30349'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D244494'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE29999'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D256374'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28703'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE29725'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28784'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28725'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE29504'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE29602'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE29284'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28933'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28956'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28880'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28978'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28775'\")\n",
      "  result = self._query(query)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chunk No. 2\n",
      "Chunk No. 3\n",
      "Chunk No. 4\n",
      "  patent_id application_number  grant_date\n",
      "0   4058839           04502532  1974-09-03\n",
      "1   3991406           04685519  1967-11-24\n",
      "2   4055746           04874760  1969-11-07\n",
      "3   4055801           05064703  1970-08-18\n",
      "4   3990710           05119535  1971-03-01\n"
     ]
    }
   ],
   "source": [
    "#Metadata Query No. 1\n",
    "metadata1 = \"SELECT patent_id, number AS application_number, date AS grant_date FROM ipinformatics.patentsview_application WHERE patent_id IN(\"\n",
    "\n",
    "count = 0\n",
    "\n",
    "for c in chunks:\n",
    "    count = count + 1\n",
    "    print(\"Chunk No. \" + str(count))\n",
    "    query = metadata1 + str(c) + \")\"\n",
    "    md1 = pd.read_sql_query(query, connection)\n",
    "    md1.set_index('patent_id')\n",
    "\n",
    "print(md1.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chunk No. 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242583'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242584'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242585'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242586'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242588'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242592'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242594'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242597'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242598'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242599'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242601'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242606'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242607'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242610'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242612'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242614'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242615'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242617'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242618'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242619'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242620'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242621'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242622'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242623'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242624'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242625'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242626'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242627'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242628'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242629'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242630'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242631'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242632'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242633'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242634'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242635'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242636'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242637'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242638'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242642'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242645'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242646'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242648'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242649'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242651'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242653'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242656'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242657'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242658'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242659'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242660'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242661'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242667'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242668'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242669'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242671'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242673'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242675'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242678'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242679'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D242682'\")\n",
      "  result = self._query(query)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chunk No. 2\n",
      "Chunk No. 3\n",
      "Chunk No. 4\n",
      "  patent_id                       assignee_id\n",
      "0   3955180  210a668b9c6902622fb733a379e9a845\n",
      "1   3956615  5866104203780349658d8543ac0f0fd4\n",
      "2   3956734  ab793877c3cb16d7e3927a856568d333\n",
      "3   3957389  ae599631c46d39900613a0a890a8289f\n",
      "4   3958081  9771095cf0c022bf34abcea8cf780232\n"
     ]
    }
   ],
   "source": [
    "#Metadata Query No. 2\n",
    "metadata2 = \"SELECT patent_id, assignee_id FROM ipinformatics.patentsview_patent_assignee WHERE patent_id IN(\"\n",
    "\n",
    "count = 0\n",
    "\n",
    "for c in chunks:\n",
    "    count = count + 1\n",
    "    print(\"Chunk No. \" + str(count))\n",
    "    query = metadata2 + str(c) + \")\"\n",
    "    md2 = pd.read_sql_query(query, connection)\n",
    "    md2.set_index('patent_id')\n",
    "    \n",
    "md2['assignee_id'] = md2.assignee_id.str.replace(r'\\r', '')\n",
    "print(md2.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chunk No. 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D028834'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D212873'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D207695'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D212416'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D209924'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D210865'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D210372'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D210373'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D211303'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D213981'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE27093'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D215723'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D218744'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D217068'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D217285'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D217966'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D216577'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D217968'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D218831'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D218209'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D218210'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D219173'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D224059'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D217928'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D217965'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D219175'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D219189'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D219361'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D219778'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D221313'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D221314'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D219118'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D221266'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D219835'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D222928'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE27643'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D192653'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D192750'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D197455'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D200042'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D198304'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D200089'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D211846'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D208580'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D207986'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D205116'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'H001519'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'PP02747'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE26887'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE26471'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE28390'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE27035'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE27321'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE26963'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE27412'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D228571'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D219463'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D220131'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D220313'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D220553'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D223183'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D220660'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D221226'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D220402'\")\n",
      "  result = self._query(query)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chunk No. 2\n",
      "Chunk No. 3\n",
      "Chunk No. 4\n",
      "  patent_id   examiner_id group_art_unit\n",
      "0   4025903             -           None\n",
      "1   4319225  DEFINED  NOT           2306\n",
      "2   3989899             -           2601\n",
      "3   4319358             -           2202\n",
      "4   4188618             -           2604\n"
     ]
    }
   ],
   "source": [
    "#Metadata Query No. 3\n",
    "metadata3 = \"SELECT patent_no AS patent_id, primary_examiner_name AS examiner_id, group_art_unit_no AS group_art_unit FROM iplogic.pair_application WHERE patent_no IN(\"\n",
    "\n",
    "count = 0\n",
    "\n",
    "for c in chunks:\n",
    "    count = count + 1\n",
    "    print(\"Chunk No. \" + str(count))\n",
    "    query = metadata3 + str(c) + \")\"\n",
    "    md3 = pd.read_sql_query(query, connection)\n",
    "    md3.set_index('patent_id')\n",
    "\n",
    "print(md3.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chunk No. 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'PP16030'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D512290'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D591464'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D519803'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D748714'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D502919'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D545404'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D600460'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D639755'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D651693'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D744479'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D539546'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D748578'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D511130'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D720980'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D515618'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D717493'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D581232'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE44760'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D503018'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D751226'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D730535'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D734898'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D519005'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D325183'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D714594'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE41885'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D507954'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D675060'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D535883'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D620851'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D270618'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D561364'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D641633'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D616148'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D714651'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D578106'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D620787'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D584625'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D525016'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D514374'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE45177'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D529481'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'PP26417'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D743752'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D714587'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE45307'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D651080'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D522609'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D695944'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D742129'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D609135'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D683795'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D615755'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE40984'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D605346'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D576581'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D667410'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'RE42015'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D567706'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D708814'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D245563'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D718369'\")\n",
      "  result = self._query(query)\n",
      "/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py:166: Warning: (1292, \"Truncated incorrect DOUBLE value: 'D539557'\")\n",
      "  result = self._query(query)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chunk No. 2\n",
      "Chunk No. 3\n",
      "Chunk No. 4\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-8-013ce9a46b89>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Chunk No. \"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcount\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0mquery\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtechclass1\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mc\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\")\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m     \u001b[0mtc1\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_sql_query\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconnection\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m     \u001b[0mtc1\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'patent_id'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pandas/io/sql.py\u001b[0m in \u001b[0;36mread_sql_query\u001b[0;34m(sql, con, index_col, coerce_float, params, parse_dates, chunksize)\u001b[0m\n\u001b[1;32m    330\u001b[0m     return pandas_sql.read_query(\n\u001b[1;32m    331\u001b[0m         \u001b[0msql\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex_col\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mindex_col\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcoerce_float\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcoerce_float\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 332\u001b[0;31m         parse_dates=parse_dates, chunksize=chunksize)\n\u001b[0m\u001b[1;32m    333\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    334\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pandas/io/sql.py\u001b[0m in \u001b[0;36mread_query\u001b[0;34m(self, sql, index_col, coerce_float, parse_dates, params, chunksize)\u001b[0m\n\u001b[1;32m   1085\u001b[0m         \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_convert_params\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msql\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1086\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1087\u001b[0;31m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexecute\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1088\u001b[0m         \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1089\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pandas/io/sql.py\u001b[0m in \u001b[0;36mexecute\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    976\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mexecute\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    977\u001b[0m         \u001b[0;34m\"\"\"Simple passthrough to SQLAlchemy connectable\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 978\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnectable\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexecute\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    979\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    980\u001b[0m     def read_table(self, table_name, index_col=None, coerce_float=True,\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/sqlalchemy/engine/base.py\u001b[0m in \u001b[0;36mexecute\u001b[0;34m(self, object, *multiparams, **params)\u001b[0m\n\u001b[1;32m    937\u001b[0m         \"\"\"\n\u001b[1;32m    938\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobject\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mutil\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstring_types\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 939\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_execute_text\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobject\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmultiparams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    940\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    941\u001b[0m             \u001b[0mmeth\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_execute_on_connection\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/sqlalchemy/engine/base.py\u001b[0m in \u001b[0;36m_execute_text\u001b[0;34m(self, statement, multiparams, params)\u001b[0m\n\u001b[1;32m   1095\u001b[0m             \u001b[0mstatement\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1096\u001b[0m             \u001b[0mparameters\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1097\u001b[0;31m             \u001b[0mstatement\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparameters\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1098\u001b[0m         )\n\u001b[1;32m   1099\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_has_events\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_has_events\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/sqlalchemy/engine/base.py\u001b[0m in \u001b[0;36m_execute_context\u001b[0;34m(self, dialect, constructor, statement, parameters, *args)\u001b[0m\n\u001b[1;32m   1187\u001b[0m                 \u001b[0mparameters\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1188\u001b[0m                 \u001b[0mcursor\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1189\u001b[0;31m                 context)\n\u001b[0m\u001b[1;32m   1190\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1191\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_has_events\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_has_events\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/sqlalchemy/engine/base.py\u001b[0m in \u001b[0;36m_handle_dbapi_exception\u001b[0;34m(self, e, statement, parameters, cursor, context)\u001b[0m\n\u001b[1;32m   1394\u001b[0m                 )\n\u001b[1;32m   1395\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1396\u001b[0;31m                 \u001b[0mutil\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreraise\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mexc_info\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1397\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1398\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/sqlalchemy/util/compat.py\u001b[0m in \u001b[0;36mreraise\u001b[0;34m(tp, value, tb, cause)\u001b[0m\n\u001b[1;32m    185\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__traceback__\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mtb\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwith_traceback\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/sqlalchemy/engine/base.py\u001b[0m in \u001b[0;36m_execute_context\u001b[0;34m(self, dialect, constructor, statement, parameters, *args)\u001b[0m\n\u001b[1;32m   1180\u001b[0m                         \u001b[0mstatement\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1181\u001b[0m                         \u001b[0mparameters\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1182\u001b[0;31m                         context)\n\u001b[0m\u001b[1;32m   1183\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mBaseException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1184\u001b[0m             self._handle_dbapi_exception(\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/sqlalchemy/engine/default.py\u001b[0m in \u001b[0;36mdo_execute\u001b[0;34m(self, cursor, statement, parameters, context)\u001b[0m\n\u001b[1;32m    468\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    469\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mdo_execute\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcursor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstatement\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparameters\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcontext\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 470\u001b[0;31m         \u001b[0mcursor\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexecute\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstatement\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparameters\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    471\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    472\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mdo_execute_no_params\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcursor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstatement\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcontext\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py\u001b[0m in \u001b[0;36mexecute\u001b[0;34m(self, query, args)\u001b[0m\n\u001b[1;32m    164\u001b[0m         \u001b[0mquery\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmogrify\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    165\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 166\u001b[0;31m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_query\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    167\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_executed\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mquery\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    168\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/cursors.py\u001b[0m in \u001b[0;36m_query\u001b[0;34m(self, q)\u001b[0m\n\u001b[1;32m    320\u001b[0m         \u001b[0mconn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_db\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    321\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_last_executed\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mq\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 322\u001b[0;31m         \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    323\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_do_get_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    324\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrowcount\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/connections.py\u001b[0m in \u001b[0;36mquery\u001b[0;34m(self, sql, unbuffered)\u001b[0m\n\u001b[1;32m    833\u001b[0m                 \u001b[0msql\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msql\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencode\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencoding\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'surrogateescape'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    834\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_execute_command\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mCOMMAND\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mCOM_QUERY\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msql\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 835\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_affected_rows\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_read_query_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0munbuffered\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0munbuffered\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    836\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_affected_rows\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    837\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/connections.py\u001b[0m in \u001b[0;36m_read_query_result\u001b[0;34m(self, unbuffered)\u001b[0m\n\u001b[1;32m   1017\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1018\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mMySQLResult\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1019\u001b[0;31m             \u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1020\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_result\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1021\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mserver_status\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/connections.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1307\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_read_load_local_packet\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfirst_packet\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1308\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1309\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_read_result_packet\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfirst_packet\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1310\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1311\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnection\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/connections.py\u001b[0m in \u001b[0;36m_read_result_packet\u001b[0;34m(self, first_packet)\u001b[0m\n\u001b[1;32m   1370\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfield_count\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfirst_packet\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_length_encoded_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1371\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_descriptions\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1372\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_read_rowdata_packet\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1373\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1374\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_read_rowdata_packet_unbuffered\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/connections.py\u001b[0m in \u001b[0;36m_read_rowdata_packet\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1404\u001b[0m         \u001b[0mrows\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1405\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1406\u001b[0;31m             \u001b[0mpacket\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnection\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_read_packet\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1407\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_packet_is_eof\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpacket\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1408\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnection\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m  \u001b[0;31m# release reference to kill cyclic reference.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/connections.py\u001b[0m in \u001b[0;36m_read_packet\u001b[0;34m(self, packet_type)\u001b[0m\n\u001b[1;32m    959\u001b[0m         \u001b[0mbuff\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34mb''\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    960\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 961\u001b[0;31m             \u001b[0mpacket_header\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_read_bytes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    962\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mDEBUG\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mdump_packet\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpacket_header\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    963\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/site-packages/pymysql/connections.py\u001b[0m in \u001b[0;36m_read_bytes\u001b[0;34m(self, num_bytes)\u001b[0m\n\u001b[1;32m    986\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    987\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 988\u001b[0;31m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_rfile\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnum_bytes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    989\u001b[0m                 \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    990\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mIOError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mOSError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/tylerseymour/anaconda/lib/python3.6/socket.py\u001b[0m in \u001b[0;36mreadinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    584\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    585\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 586\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    587\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    588\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_timeout_occurred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "#Tech Class 1 - Query No. 4\n",
    "techclass1 = \"SELECT patent_id, mainclass_id AS uspc, sequence FROM ipinformatics.patentsview_uspc WHERE patent_id IN(\"\n",
    "\n",
    "count = 0\n",
    "\n",
    "for c in chunks:\n",
    "    count = count + 1\n",
    "    print(\"Chunk No. \" + str(count))\n",
    "    query = techclass1 + str(c) + \")\"\n",
    "    tc1 = pd.read_sql_query(query, connection)\n",
    "    tc1.set_index('patent_id')\n",
    "\n",
    "print(tc1.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Tech Class 2 - Query No. 5\n",
    "\n",
    "techclass2 = \"SELECT patent_id, group_id AS cpc, sequence FROM ipinformatics.patentsview_cpc_current WHERE patent_id IN(\"\n",
    "\n",
    "count = 0\n",
    "\n",
    "for c in chunks:\n",
    "    count = count + 1\n",
    "    print(\"Chunk No. \" + str(count))\n",
    "    query = techclass2 + str(c) + \")\"\n",
    "    tc2 = pd.read_sql_query(query, connection)\n",
    "    tc2.set_index('patent_id')\n",
    "\n",
    "print(tc2.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Citations 1 - Query No. 6\n",
    "cites1 = \"SELECT patent_id, citation_id, date, category, sequence FROM ipinformatics.patentsview_us_patent_citation WHERE patent_id IN(\"\n",
    "\n",
    "count = 0\n",
    "\n",
    "for c in chunks:\n",
    "    count = count + 1\n",
    "    print(\"Chunk No. \" + str(count))\n",
    "    query = cites1 + str(c) + \")\"\n",
    "    ct1 = pd.read_sql_query(query, connection)\n",
    "    ct1.set_index('patent_id')\n",
    "\n",
    "print(ct1.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total chunks to do: 5\n",
      "Size of remainder: 2117\n",
      "Chunk No. 1\n",
      "Chunk No. 2\n",
      "Chunk No. 3\n",
      "Chunk No. 4\n",
      "  patent_id                                               text  sequence  \\\n",
      "0   3955073  21. The device of claim 17 further including:m...        21   \n",
      "1   3955073  5. The device of claim 4 wherein said decimal ...         5   \n",
      "2   3955073  12. The device of claim 1 wherein said sensing...        12   \n",
      "3   3955073  22. A device for conveniently determining and ...        22   \n",
      "4   3955073  3. The device of claim 2 wherein said mechanic...         3   \n",
      "\n",
      "   dependent  \n",
      "0          0  \n",
      "1          0  \n",
      "2          0  \n",
      "3          0  \n",
      "4          0  \n"
     ]
    }
   ],
   "source": [
    "#Claims 1 - Query No. 7\n",
    "\n",
    "#chunker\n",
    "chunks = []\n",
    "subchunk = []\n",
    "\n",
    "maxchunks = int(len(idlist)/5000)\n",
    "maxchunks_real = maxchunks + 1\n",
    "print(\"Total chunks to do: \" + str(maxchunks_real))\n",
    "\n",
    "remainder = len(idlist) - (maxchunks*5000)\n",
    "print(\"Size of remainder: \" + str(remainder))\n",
    "\n",
    "count = 0\n",
    "count2 = 0\n",
    "\n",
    "for id in idlist:\n",
    "    subchunk.append(id)\n",
    "    if count <= maxchunks:\n",
    "        if len(subchunk) == 5000:\n",
    "            yy = \"','\".join(str(s) for s in subchunk)\n",
    "            chunks.append(yy)\n",
    "            count = count + 1\n",
    "            subchunk = []\n",
    "    elif count > maxchunks:\n",
    "        if len(subchunk) == remainder:\n",
    "            yy = \"','\".join(str(s) for s in subchunk)\n",
    "            chunks.append(yy)\n",
    "            count = count + 1\n",
    "            subchunk = []\n",
    "claims1 = \"SELECT patent_id, text, sequence, dependent FROM ipinformatics.patentsview_claim WHERE patent_id IN('\"\n",
    "\n",
    "count = 0\n",
    "\n",
    "for c in chunks:\n",
    "    count = count + 1\n",
    "    print(\"Chunk No. \" + str(count))\n",
    "    query = claims1 + str(c) + \"')\"\n",
    "    clm1 = pd.read_sql_query(query, connection)\n",
    "    clm1.set_index('patent_id')\n",
    "\n",
    "print(clm1.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'md1' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-1d1ee11eda2e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m#Ultimate Merge\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mmerge1\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmd1\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmd2\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'patent_id'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'left'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mmerge2\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmerge1\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmd3\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'patent_id'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'left'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mmerge3\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmerge2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtc1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'patent_id'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'left'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mfinalmerge\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmerge3\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mct1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'patent_id'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'left'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'md1' is not defined"
     ]
    }
   ],
   "source": [
    "#Ultimate Merge\n",
    "merge1 = md1.merge(md2, on='patent_id', how='left')\n",
    "merge2 = merge1.merge(md3, on='patent_id', how='left')\n",
    "merge3 = merge2.merge(tc1, on='patent_id', how='left')\n",
    "finalmerge = merge3.merge(ct1, on='patent_id', how='left')\n",
    "\n",
    "print(finalmerge.head())\n",
    "print(\"Number of records for this query -->\")\n",
    "print(finalmerge.size)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  patent_id application_number  grant_date                       assignee_id  \\\n",
      "0   4058839           04502532  1974-09-03  c40d37876fb270067e0369462e15daba   \n",
      "1   4058839           04502532  1974-09-03  c40d37876fb270067e0369462e15daba   \n",
      "2   4058839           04502532  1974-09-03  c40d37876fb270067e0369462e15daba   \n",
      "3   4058839           04502532  1974-09-03  c40d37876fb270067e0369462e15daba   \n",
      "4   4058839           04502532  1974-09-03  c40d37876fb270067e0369462e15daba   \n",
      "\n",
      "  examiner_id group_art_unit uspc  sequence_x citation_id        date  \\\n",
      "0         NaN            NaN  235         2.0     3308010  1967-03-01   \n",
      "1         NaN            NaN  235         2.0     3308010  1967-03-01   \n",
      "2         NaN            NaN  235         2.0     3308010  1967-03-01   \n",
      "3         NaN            NaN  235         2.0     3308010  1967-03-01   \n",
      "4         NaN            NaN  235         2.0     3308010  1967-03-01   \n",
      "\n",
      "  category  sequence_y                                               text  \\\n",
      "0                  1.0  1. A flexible resilient magnetic particle type...   \n",
      "1                  1.0  12. An information card in accordance with cla...   \n",
      "2                  1.0  18. An information card in accordance with cla...   \n",
      "3                  1.0  21. An information card in accordance with cla...   \n",
      "4                  1.0  20. An information card in accordance with cla...   \n",
      "\n",
      "   sequence  dependent  \n",
      "0         1          0  \n",
      "1        12          0  \n",
      "2        18          0  \n",
      "3        21          0  \n",
      "4        20          0  \n"
     ]
    }
   ],
   "source": [
    "#Switching to new dataframe DF....\n",
    "df = finalmerge\n",
    "print(df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  patent_id application_number  grant_date                       assignee_id  \\\n",
      "0   4058839           04502532  1974-09-03  c40d37876fb270067e0369462e15daba   \n",
      "1   4058839           04502532  1974-09-03  c40d37876fb270067e0369462e15daba   \n",
      "2   4058839           04502532  1974-09-03  c40d37876fb270067e0369462e15daba   \n",
      "3   4058839           04502532  1974-09-03  c40d37876fb270067e0369462e15daba   \n",
      "4   4058839           04502532  1974-09-03  c40d37876fb270067e0369462e15daba   \n",
      "\n",
      "  examiner_id group_art_unit uspc  sequence_x citation_id        date  \\\n",
      "0         NaN            NaN  235         2.0     3308010  1967-03-01   \n",
      "1         NaN            NaN  235         2.0     3308010  1967-03-01   \n",
      "2         NaN            NaN  235         2.0     3308010  1967-03-01   \n",
      "3         NaN            NaN  235         2.0     3308010  1967-03-01   \n",
      "4         NaN            NaN  235         2.0     3308010  1967-03-01   \n",
      "\n",
      "  category  sequence_y                                               text  \\\n",
      "0                  1.0  1. A flexible resilient magnetic particle type...   \n",
      "1                  1.0  12. An information card in accordance with cla...   \n",
      "2                  1.0  18. An information card in accordance with cla...   \n",
      "3                  1.0  21. An information card in accordance with cla...   \n",
      "4                  1.0  20. An information card in accordance with cla...   \n",
      "\n",
      "   sequence  dependent  \n",
      "0         1          0  \n",
      "1        12          0  \n",
      "2        18          0  \n",
      "3        21          0  \n",
      "4        20          0  \n"
     ]
    }
   ],
   "source": [
    "print(df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-0d9e5978e3d9>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_json\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"ultrarip.json\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morient\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'records'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlines\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'df' is not defined"
     ]
    }
   ],
   "source": [
    "df.to_json(\"ultrarip.json\", orient='records', lines=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
